modules/apps/remote-web-app/remote-web-app-web/src/main/java/com/liferay/remote/web/app/web/internal/portlet/RemoteWebAppPortlet.java

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.remote.web.app.web.internal.portlet;
+
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCPortlet;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.util.ResourceBundleLoader;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+
+import java.util.Collections;
+import java.util.Dictionary;
+import java.util.Enumeration;
+import java.util.Hashtable;
+import java.util.Locale;
+import java.util.Map;
+import java.util.ResourceBundle;
+
+import javax.portlet.Portlet;
+import javax.portlet.RenderRequest;
+import javax.portlet.RenderResponse;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+
+/**
+ * @author Iván Zaera Avellón
+ */
+public class RemoteWebAppPortlet extends MVCPortlet {
+
+	public RemoteWebAppPortlet(long id, String name, String url) {
+		_id = id;
+		_name = name;
+		_url = url;
+	}
+
+	public String getName() {
+		return _name;
+	}
+
+	public synchronized void register(BundleContext bundleContext) {
+		if (_serviceRegistration != null) {
+			throw new IllegalStateException("Portlet is already registered");
+		}
+
+		Dictionary<String, Object> properties = new Hashtable<>();
+
+		properties.put(
+			"com.liferay.portlet.display-category", "category.sample");
+		properties.put("com.liferay.portlet.instanceable", true);
+		properties.put("javax.portlet.name", _getPortletName());
+		properties.put("javax.portlet.security-role-ref", "power-user,user");
+		properties.put(
+			"javax.portlet.resource-bundle", _getResourceBundleName());
+
+		_serviceRegistration = bundleContext.registerService(
+			Portlet.class, this, properties);
+
+		properties = new Hashtable<>();
+
+		properties.put("resource.bundle.base.name", _getResourceBundleName());
+		properties.put("servlet.context.name", "remote-web-app-web");
+
+		_resourceBundleLoaderServiceRegistration =
```

Comment:
Register a `ResourceBundleLoader` to emulate the equivalent of `content.Language.properties` in portlet bundles. 

Note that we register it for the servlet context name, so that the portlet title can be retrieved by the architecture. 

We may need to register more auxiliary services in the future given that dynamic portlets all belong to the same bundle (remote-web-app-web).